### PR TITLE
add doc about migrations

### PR DESCRIPTION
**Pull request by craig-day**
Some updates and clarifications to the readme. This also adds a section on how to deal with migrations. 

/cc @zendesk/sustaining @grosser @pschambacher @bquorning

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,15 @@ switch between database connections. We've made the implementation very small, a
 
 ActiveRecord Shards has been used and tested on Rails 3.2, 4.2 and 5.0 and has in some form or another been used in production on a large Rails app for several years.
 
+- [Installation](#installation)
+- [Configuration](#configuration)
+- [Migrations](#migrations)
+  - [Example](#example)
+    - [Shared Model](#create-a-table-for-the-shared-not-sharded-model)
```
Comment (grosser) [inline README.md]: maybe we should call this `global` or so to make it less likely to be confused visually `shared` <-> `sharded`

Comment (craig-day) [reply README.md]: Yeah, maybe a followup PR. Stuck with the existing naming to be consistent.

**Comment by craig-day** [comment]
I changed the readme... idk how I possible broke every test

**Comment by bquorning** [comment]
I have a fix for the failing tests. Somewhere. Hang on.

**Pull request merged**